### PR TITLE
Alter multiple scram/tls users using UO

### PR DESCRIPTION
**Pull request by michalxo**
Signed-off-by: Michal Tóth <mtoth@redhat.com>

New test for #5305 (also added for TLS as it's tight together)

Other suggested tests for SCRAM are in UserST, SecurityST, ListenersST and MirrorMaker2ST or HttpBridgeScramShaST for example.

- [x] Write tests
- [x] Make sure all tests pass
- [x] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 1]
```diff
@@ -158,7 +159,23 @@ void testBigAmountOfScramShaUsers(ExtensionContext extensionContext) {
         String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
         String userName = mapWithTestUsers.get(extensionContext.getDisplayName());
 
-        createBigAmountOfUsers(extensionContext, userName, "SCRAM_SHA");
+        createBigAmountOfUsers(extensionContext, userName, "SCRAM_SHA", 100);
+    }
+
+    @Tag(SCALABILITY)
+    @ParallelTest
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:166]: Should these kind of tests really run in parallel?

Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:166]: I am not creator of them, just this one, but why not? 
It stresses UO a bit more. IIUC single kafka cluster is being used for all of these tests.

But maybe @see-quick or @im-konge may share/see your concern?

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=UserST

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 1 above)
Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:166]: I think the idea of running load tests in parallel on a shared environment is. bit weird ... btu lets see how it works 🤷‍♂️

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=UserST

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 2]
```diff
@@ -158,7 +159,25 @@ void testBigAmountOfScramShaUsers(ExtensionContext extensionContext) {
...
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(NAMESPACE, extensionContext);
+        String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        String userName = mapWithTestUsers.get(extensionContext.getDisplayName());
+        int numberOfUsers = 100;
+        int producerRate = 1000;
+        int consumerRate = 1500;
+        int requestsPercentage = 42;
+        double mutationRate = 3.0;
+
+        deployKafka(extensionContext, clusterName);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:177]: why do you need a second kafka?

**Comment by michalxo** [comment]
```
[WARNING] io.strimzi.systemtest.operators.user.UserST.testBigAmountOfTlsUsers(ExtensionContext)
[ERROR]   Run 1: UserST.testBigAmountOfTlsUsers:189->createBigAmountOfUsers:411 ? Wait Timeout ...
[INFO]   Run 2: PASS
[INFO]
[WARNING] io.strimzi.systemtest.operators.user.UserST.testScramUserWithQuotas(ExtensionContext)
[ERROR]   Run 1: UserST.testScramUserWithQuotas:221 ? Wait Timeout after 180000 ms waiting for ...
[INFO]   Run 2: PASS
[INFO]
[WARNING] io.strimzi.systemtest.operators.user.UserST.testTlsUserWithQuotas(ExtensionContext)
[ERROR]   Run 1: UserST.testTlsUserWithQuotas:214 ? Wait Timeout after 180000 ms waiting for Re...
[INFO]   Run 2: PASS
[INFO]
[WARNING] io.strimzi.systemtest.operators.user.UserST.testUserWithNameMoreThan64Chars(ExtensionContext)
[ERROR]   Run 1: UserST.testUserWithNameMoreThan64Chars:69 ? Wait Timeout after 180000 ms waiti...
[INFO]   Run 2: PASS
[INFO]
[INFO]
[WARNING] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Flakes: 4
```

**Comment by Frawless** [comment]
> ```
> [WARNING] io.strimzi.systemtest.operators.user.UserST.testBigAmountOfTlsUsers(ExtensionContext)
> [ERROR]   Run 1: UserST.testBigAmountOfTlsUsers:189->createBigAmountOfUsers:411 ? Wait Timeout ...
> [INFO]   Run 2: PASS
> [INFO]
> [WARNING] io.strimzi.systemtest.operators.user.UserST.testScramUserWithQuotas(ExtensionContext)
> [ERROR]   Run 1: UserST.testScramUserWithQuotas:221 ? Wait Timeout after 180000 ms waiting for ...
> [INFO]   Run 2: PASS
> [INFO]
> [WARNING] io.strimzi.systemtest.operators.user.UserST.testTlsUserWithQuotas(ExtensionContext)
> [ERROR]   Run 1: UserST.testTlsUserWithQuotas:214 ? Wait Timeout after 180000 ms waiting for Re...
> [INFO]   Run 2: PASS
> [INFO]
> [WARNING] io.strimzi.systemtest.operators.user.UserST.testUserWithNameMoreThan64Chars(ExtensionContext)
> [ERROR]   Run 1: UserST.testUserWithNameMoreThan64Chars:69 ? Wait Timeout after 180000 ms waiti...
> [INFO]   Run 2: PASS
> [INFO]
> [INFO]
> [WARNING] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Flakes: 4
> ```

I think you should fix the flakes

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 2 above)
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java]: Trying to fix the tests - no more use of 2 kafkas. Moved to isolatedTests approach. Still seeing issue with UserOperator. Talking with our developers about this. There might be a bug in UO..

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=UserST

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=UserST

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=UserST

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 3]
```diff
@@ -369,34 +394,78 @@ void testCreatingUsersWithSecretPrefix(ExtensionContext extensionContext) {
...
             }
 
             LOGGER.info("Checking status of KafkaUser {}", userNameWithSuffix);
-            Condition kafkaCondition = KafkaUserResource.kafkaUserClient().inNamespace(NAMESPACE).withName(userNameWithSuffix).get()
+            Condition kafkaCondition = KafkaUserResource.kafkaUserClient().inNamespace(namespace).withName(userNameWithSuffix).get()
+                    .getStatus().getConditions().get(0);
+            LOGGER.info("KafkaUser condition status: {}", kafkaCondition.getStatus());
+            LOGGER.info("KafkaUser condition type: {}", kafkaCondition.getType());
+            assertThat(kafkaCondition.getType(), is(Ready.toString()));
+            LOGGER.info("KafkaUser {} is in desired state: {}", userNameWithSuffix, kafkaCondition.getType());
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:421]: I don't think that we want statuses of all users, created in this method, to be printed into `INFO` level.

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 4]
```diff
@@ -153,38 +155,61 @@ void testUpdateUser(ExtensionContext extensionContext) {
     }
 
     @Tag(SCALABILITY)
-    @ParallelTest
+    @IsolatedTest
```
[the remark this answers is not in the document]
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:158]: Yes, that was my intention - to offload kafka/cluster.

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 5]
```diff
@@ -369,34 +394,78 @@ void testCreatingUsersWithSecretPrefix(ExtensionContext extensionContext) {
         assertNull(kubeClient(namespaceName).getSecret(scramShaSecret.getMetadata().getName()));
     }
 
-    synchronized void createBigAmountOfUsers(ExtensionContext extensionContext, String userName, String typeOfUser) {
-
-        int numberOfUsers = 100;
-
+    synchronized void createBigAmountOfUsers(ExtensionContext extensionContext, String namespace, String clusterName, String userName, String typeOfUser, int numberOfUsers) {
```
[the remark this answers is not in the document]
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:397]: Thanks, good catch. This is a remnant of my parallel tests approach.

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 3 above)
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:421]: That's not my code, but I will change it to debug.

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:421]: Thanks

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=UserST

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 6]
```diff
@@ -465,10 +491,57 @@ synchronized void createBigAmountOfUsers(ExtensionContext extensionContext, Stri
...
+                    .editMetadata()
+                        .withNamespace(NAMESPACE)
+                    .endMetadata()
+                        .editSpec()
+                            .withQuotas(kuq)
+                        .endSpec()
+                    .build());
+            }
+
+            LOGGER.info("[After update] Checking status of KafkaUser {}", userNameWithSuffix);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:531]: I wonder if we really need this useless log on INFO level when details are on DEBUG level?

Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:531]: You can wonder the same for few lines above - with user creation - same story.
Let me know what you figured out. ;-)

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:531]: Move ti to DEBUG level and just log info before loops

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=UserST

**Review by im-konge — APPROVED** [review]

**Comment by Frawless** [comment]
@scholzj do you want to do one final review or should we merge it? We approved it with @im-konge  already

**Review by scholzj — APPROVED** [review]

**Comment by michalxo** [comment]
Issue found in regressions is the one I mentioned earlier about timeouts in user creation of UserOperator.
 @ShubhamRwt is looking into it (and managed to reproduce it)

**Comment by scholzj** [comment]
@michalxo But we cannot merge something what breaks the regression tests.

**Comment by scholzj** [comment]
So I guess what you can do is either move it to Draft and wait for the underlying issue to be fixed or disable the failing tests with a comment linking it to the GitHub issue about the problem so that it can be re-enabled later.

**Comment by michalxo** [comment]
Yes, I don't want to merge it until fixed (all tests are passing). 
Moving to draft then, until UO issue is fixed.

**Comment by scholzj** [comment]
> Yes, I don't want to merge it until fixed (all tests are passing).
Moving to draft then, until UO issue is fixed.

I wonder if the other variant is better as the fix might not come soon. But ok, lets keep it as Draft.

**Comment by Frawless** [comment]
+1 for disable the tests and merge the PR. @ShubhamRwt then can easily verify the fix automatically.

**Comment by michalxo** [comment]
Disabled all four tests, which create many users (any of these tests can fail any time).
Ready for merge (I suppose).

**Pull request merged**